nsv13/code/modules/munitions/ammunition/torpedos/torpedo_construction.dm

```diff
@@ -0,0 +1,301 @@
+		if(7)
+			. += "<span class='notice'>The warhead is sitting snug at the nose end of the casing. The bolts could be tighter.</span>"
+		if(8)
+			. += "<span class='notice'>The casing contains the warhead, an IFF chip, guidance and propulsion systems. They are not yet wired together.</span>"
+		if(9)
+			. += "<span class='notice'>The casing has the following components installed: [wh?.name], [iff?.name], [gs?.name], [ps?.name]. It looks ready to close and bolt shut. </span>"
+		if(10)
+			. += "<span class='notice'>The casing has been closed and bolted shut. It only requires sealing with a welding tool to be ready for action.</span>"
+
+/obj/item/ship_weapon/ammunition/torpedo/torpedo_casing/attackby(obj/item/W, mob/user, params)
+	add_fingerprint(user)
+	if(istype(W, /obj/item/ship_weapon/parts/torpedo/warhead))
+		if(state == 6)
+			to_chat(user, "<span class='notice'>You start adding [W] to [src]...</span>")
+			if(!do_after(user, 2 SECONDS, target=src))
+				return
+			to_chat(user, "<span class='notice'>You add [W] to [src].</span>")
+			wh = W
+			state = 7
+			update_icon()
+			W.forceMove(src)
+		return
+	else if(istype(W, /obj/item/ship_weapon/parts/torpedo/guidance_system))
+		if(state == 2)
+			to_chat(user, "<span class='notice'>You start adding [W] to [src]...</span>")
+			if(!do_after(user, 2 SECONDS, target=src))
+				return
+			to_chat(user, "<span class='notice'>You add [W] to [src].</span>")
+			gs = W
+			state = 3
+			update_icon()
+			W.forceMove(src)
+		return
+	else if(istype(W, /obj/item/ship_weapon/parts/torpedo/propulsion_system))
+		if(state == 0)
+			to_chat(user, "<span class='notice'>You start adding [W] to [src]...</span>")
+			if(!do_after(user, 2 SECONDS, target=src))
+				return
+			to_chat(user, "<span class='notice'>You add [W] to [src].</span>")
+			ps = W
+			state = 1
+			update_icon()
+			W.forceMove(src)
+		return
+	else if(istype(W, /obj/item/ship_weapon/parts/torpedo/iff_card))
+		if(state == 4)
+			to_chat(user, "<span class='notice'>You start adding [W] to [src]...</span>")
+			if(!do_after(user, 2 SECONDS, target=src))
+				return
+			to_chat(user, "<span class='notice'>You add [W] to [src].</span>")
+			iff = W
+			state = 5
+			update_icon()
+			W.forceMove(src)
+		return
+	else if(istype(W, /obj/item/stack/cable_coil))
+		var/obj/item/stack/cable_coil/C = W
+		if(state == 8)
+			if(C.get_amount() < 3)
+				to_chat(user, "<span class='notice'>You need at least three cable pieces to wire [src]!</span>") //for 'realistic' wire spaghetti
+				return
+			to_chat(user, "<span class='notice'>You start wiring [src]...</span>")
+			if(!do_after(user, 2 SECONDS, target=src))
+				return
+			W.use(3)
+			to_chat(user, "<span class='notice'>You wire [src].</span>")
+			state = 9
+			update_icon()
+		return
+
+/obj/item/ship_weapon/ammunition/torpedo/torpedo_casing/wrench_act(mob/user, obj/item/tool)
+	. = FALSE
```

Comment:
?

Reply:
Dunno. Updating to call parent at the end instead.